### PR TITLE
[DOCS-1605] Optional license, new image for k8s db-backed installation

### PR DESCRIPTION
**Pull request by lena-larionova**
### Review

@cwurm, @rainest, @Bradamant3 

### Summary
Updating the installation doc for Kong Gateway (Enterprise) on Kubernetes (final name TBD).

**Note to reviewers:** I only updated db-backed install topic, as the db-less topic uses a YAML manifest and those are not yet updated: https://github.com/Kong/kubernetes-ingress-controller/issues/1041

* Removing mention of bintray licenses - users should only get them from sales rep at the moment
* Went with suggested dummy license secret route - @rainest Ideally we could get a license-less chart and let people patch it in the future, but that's also not currently set up?
* General topic clean up and consistency
* Labelling Dev Portal as enterprise/optional feature throughout
* Use public `kong-docker-kong-gateway-docker.bintray.io/kong-enterprise-edition` instead of private `kong-docker-kong-enterprise-edition-docker.bintray.io/kong-enterprise-edition`, and remove Docker registry login secret.

### Reason
Kong Gateway on Kubernetes is available in free mode.

### Testing
https://deploy-preview-2670--kongdocs.netlify.app/enterprise/2.3.x/deployment/installation/kong-on-kubernetes/

### CODE REVIEW

**Comment by rainest** [comment]
Optional license support in the chart should be available soon-ish https://github.com/Kong/charts/pull/310

File: app/enterprise/2.3.x/deployment/installation/kong-on-kubernetes.md [hunk 1]
```diff
@@ -57,119 +53,152 @@ $ oc new-project kong
...
 {% endnavtab %}
 {% endnavtabs %}
+<!-- end of codeblock navtabs -->
+
 <div class="alert alert-warning">
 <i class="fas fa-exclamation-triangle" style="color:orange; margin-right:3px"></i>
 <strong>Important:</strong>Though not required, this is recommended if you want to use RBAC, as it cannot be done after initial setup.</div>
 
-## Step 6. Prepare the sessions plugin for Kong Manager and Dev Portal
+## Step 5. Prepare the sessions plugin for Kong Manager and Dev Portal
```
Comment (cwurm) [inline app/enterprise/2.3.x/deployment/installation/kong-on-kubernetes.md:145]: With this now Step 5, the subsequent step numbers need to be adjusted as well.

**Review by cwurm — APPROVED** [review]
Looks good, really like the formatting fixes as well. Just need to readjust the later step numbers.

**Review by HCloward — APPROVED** [review]
LGTM!

**Pull request merged**